src/main/java/io/quarkus/gizmo/ClassCreator.java

```diff
@@ -73,6 +73,14 @@ public ClassCreator(ClassOutput classOutput, String name, String signature, Stri
         this(null, classOutput, name, signature, superClass, 0, interfaces);
     }
 
+    public MethodCreator getConstructor(String... parameters) {
+        return getMethodCreator(MethodDescriptor.ofConstructor(className, parameters));
+    }
+
+    public MethodCreator getSuperConstructor(String... parameters) {
```

Comment:
This does not seem right, you can't create the superclasses constructor in a subclass.

Reply:
hmm you are right. Indeed I start this one then switch to returnValue patch to implement the call super constructor patch. I think this one was just to delete. I will close it.